src/Domain/FunctionsOfTime/SettleToConstantQuaternion.cpp

```diff
@@ -0,0 +1,136 @@
+                  gsl::at(unnormalized_func_and_derivs, 2)) /
+          gsl::at(result, 0);
+    }
+  }
+  return result;
+}
+};  // namespace
+
+namespace domain::FunctionsOfTime {
+SettleToConstantQuaternion::SettleToConstantQuaternion(
+    const std::array<DataVector, 3>& initial_func_and_derivs,
+    const double match_time, const double decay_time)
+    : unnormalized_function_of_time_(initial_func_and_derivs, match_time,
+                                     decay_time),
+      match_time_(match_time),
+      decay_time_(decay_time),
+      size_(initial_func_and_derivs[0].size()) {
+  if (size_ != 4) {
+    ERROR(
+        "SettleToConstantQuaternion requires the initial function and its time "
+        "derivatives to be quaternions stored as DataVectors of size 4, not "
+        "size "
+        << size_);
+  }
+}
+
+std::unique_ptr<FunctionOfTime> SettleToConstantQuaternion::get_clone() const {
+  return std::make_unique<SettleToConstantQuaternion>(*this);
+}
+
+template <size_t MaxDerivReturned>
+std::array<DataVector, MaxDerivReturned + 1>
+SettleToConstantQuaternion::func_and_derivs(const double t) const {
+  static_assert(MaxDerivReturned < 3, "The maximum available derivative is 2.");
+
+  std::array<DataVector, MaxDerivReturned + 1> result =
+      make_array<MaxDerivReturned + 1>(DataVector(size_, 0.0));
+  if constexpr (MaxDerivReturned == 0) {
+    result = unnormalized_function_of_time_.func(t);
+  } else if constexpr (MaxDerivReturned == 1) {
+    result = unnormalized_function_of_time_.func_and_deriv(t);
+  } else if constexpr (MaxDerivReturned == 2) {
+    result = unnormalized_function_of_time_.func_and_2_derivs(t);
+  }
+  std::array<double, MaxDerivReturned + 1> norm_and_derivs =
+      quat_norm_and_derivs<MaxDerivReturned>(result);
+  gsl::at(result, 0) /= gsl::at(norm_and_derivs, 0);
+  if constexpr (MaxDerivReturned > 0) {
+    gsl::at(result, 1) /= gsl::at(norm_and_derivs, 0);
+    gsl::at(result, 1) -= gsl::at(result, 0) * gsl::at(norm_and_derivs, 1) /
+                          gsl::at(norm_and_derivs, 0);
+    if constexpr (MaxDerivReturned > 1) {
+      gsl::at(result, 2) /= gsl::at(norm_and_derivs, 0);
+      gsl::at(result, 2) -= gsl::at(result, 1) * 2.0 *
+                            gsl::at(norm_and_derivs, 1) /
+                            gsl::at(norm_and_derivs, 0);
+      gsl::at(result, 2) -= gsl::at(result, 0) * gsl::at(norm_and_derivs, 2) /
+                            gsl::at(norm_and_derivs, 0);
+    }
+  }
+
+  return result;
+}
+
+void SettleToConstantQuaternion::pup(PUP::er& p) {
+  FunctionOfTime::pup(p);
+  size_t version = 0;
+  p | version;
+  // Remember to increment the version number when making changes to this
+  // function. Retain support for unpacking data written by previous versions
+  // whenever possible. See `Domain` docs for details.
+  if (version >= 0) {
+    p | unnormalized_function_of_time_;
+    p | match_time_;
+    p | decay_time_;
```

Comment:
If you choose to keep the `size_` variable, then you'll have to either pup it, or set it from `unnormalized_function_of_time_`. However, if you get rid of `size_` then there's nothing to do.

Reply:
I got rid of it